src/zenml/utils/yaml_utils.py

```diff
@@ -142,6 +149,7 @@ def write_json(
         file_path,
         json.dumps(
             contents,
+            ensure_ascii=not ZENML_ALLOW_NON_ASCII_MATERIALIZERS_DUMPS,
```

Comment:
With the current implementation and environment variable name, I think this is a little misleading. The environment variable signals that it affects just the materializer, while it actually changes the behaviour of any json file that ZenML is writing.

I think there are two solutions:
- We change the name of the environment variable to reflect what it actually does
- We update the code and only pass this when used by the materializer. This could be done by for example adding `**json_dump_args` to the `write_json` function which get forwarded to the `json.dumps` call.